pkg/did/creator/creator.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright Avast Software. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package creator defines a type that can be used to create DIDs using various supported DID methods.
+package creator
+
+import (
+	"crypto/ed25519"
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+
+	"github.com/trustbloc/wallet-sdk/pkg/api"
+	didkeycreator "github.com/trustbloc/wallet-sdk/pkg/did/key"
+)
+
+const (
+	// DIDMethodKey is the name used by the Create method for the did:key method.
+	DIDMethodKey = "key"
+
+	didMethodPrefix            = "did:"
+	ed25519VerificationKey2018 = "Ed25519VerificationKey2018"
+)
+
+// DIDCreator is used for creating DID Documents using supported DID methods.
+type DIDCreator struct {
+	keyHandleReader api.KeyHandleReader
+}
+
+// NewDIDCreator returns a new DIDCreator.
+// keyHandleReader is optional. If provided, then it will be used to fetch a key (based on the KeyID in the options)
+// in the Create method. If not provided, then a new key will be generated and used for DID creation.
+func NewDIDCreator(keyHandleReader api.KeyHandleReader) *DIDCreator {
+	return &DIDCreator{
+		keyHandleReader: keyHandleReader,
+	}
+}
+
+// Create creates a DID document for the given DID method.
+// The usage of createDIDOpts depends on the DID method you're using.
+func (d *DIDCreator) Create(method string, createDIDOpts *api.CreateDIDOpts) (*did.DocResolution, error) {
+	if method == DIDMethodKey || method == fmt.Sprintf("%s%s", didMethodPrefix, DIDMethodKey) {
+		return d.createKeyDIDDoc(createDIDOpts)
+	}
+
+	return nil, fmt.Errorf("DID method %s not supported", method)
+}
+
+func (d *DIDCreator) createKeyDIDDoc(createDIDOpts *api.CreateDIDOpts) (*did.DocResolution, error) {
+	var key []byte
+
+	var keyType string
+
+	if d.keyHandleReader == nil {
```

Comment:
Can you explain this?

When client provided `kid` then you simply read from key reader & if key reader not setup then throw error.

By default, you create a keypair in keyhandle writer using key type provided.
Use defauult ED25519 as keytype.

Reply:
What the code here does is, if the key reader is set up, then I try reading from there. Otherwise, I generate a new key for them and use that for the did document creation. I'll update it so that if the keyID is used but no key reader is set up, then an error gets thrown. I haven't added a keyHandleCreator yet due to the reason in my other comment (will discuss with you separately, as that one is a bit more complicated)